### PR TITLE
update codec to 2.0

### PR DESCRIPTION
**Pull request by thiolliere**
Can we release a new version of finality-grandpa with these change ? This is needed for substrate.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -12,9 +12,9 @@ futures = { version = "0.3.5", default-features = false }
 futures-timer = { version = "3.0.2", optional = true }
 log = { version = "0.4", optional = true }
 parking_lot = { version = "0.11", optional = true }
-parity-scale-codec = { version = "1.3.5", default-features = false, features = ["derive"], optional = true }
+parity-scale-codec = { version = "2.0.0", default-features = false, features = ["derive"], optional = true }
 num = { package = "num-traits", version = "0.2", default-features = false }
-rand = { version = "0.7.0", optional = true }
+rand = { version = "0.8.0", optional = true }
```
Comment (thiolliere) [inline Cargo.toml:17]: I just checked that it compiles

File: src/fuzz_helpers.rs [hunk 2]
```diff
@@ -407,7 +407,7 @@ pub fn execute_fuzzed_graph(data: &[u8]) {
 
 	/// A vote-node on the graph.
 	#[derive(Default, Clone, Debug)]
-	struct Vote { prevote: u8, precommit: u8 };
+	struct Vote { prevote: u8, precommit: u8 }
```
Comment (thiolliere) [inline src/fuzz_helpers.rs:410]: this is just an unnecessary trailing semicolon

**Comment by andresilva** [comment]
@thiolliere Does it change the underlying encoding? Asking because it's easiest to do backport and do a minor release after.

**Review by andresilva — APPROVED** [review]

**Comment by andresilva** [comment]
@thiolliere Nevermind, I will just release a new major.

**Comment by thiolliere** [comment]
> @thiolliere Does it change the binary encoding? Asking because it's easiest to backport and do a minor release after.

no it doesn't change any binary encoding, but wouldn't the backport be a breaking change anyway ? as libraries using it would loose the implementation of Encode/Decode for parity-scale-code 1.0 no ?

**Comment by andresilva** [comment]
Yeah, you are right. No worries, I checked and there's nothing new in master besides updated dependencies, everything else was backported already to v0.12, so there's no problem with releasing v0.13.

**Comment by andresilva** [comment]
Merging before CI finishes since Travis is taking forever to run.

**Pull request merged**